### PR TITLE
Decouple controller_for test from globals resolver

### PR DESCRIPTION
**Pull request by stevenwu**
Thanks @mmun for the help.

#15058

### CODE REVIEW

File: packages/ember-routing/tests/system/controller_for_test.js [hunk 1]
```diff
@@ -1,112 +1,59 @@
...
+    let MainController = Controller.extend();
+    this.add('controller:basic', MainController);
 
-  ok(controller instanceof Controller, 'should return controller');
-});
-
-
-QUnit.test('generateController should return App.Controller if provided', function() {
-  let controller;
-  namespace.Controller = Controller.extend();
```
Comment (mmun) [inline packages/ember-routing/tests/system/controller_for_test.js]: This line became https://github.com/emberjs/ember.js/pull/15100/files#diff-5251a48efdb95856746f72151cc33bbbR32, but I would have expected the factory be registered as `controller:main` and not `controller:basic`.

Comment (mixonic) [inline packages/ember-routing/tests/system/controller_for_test.js]: Seems like it is `basic` https://github.com/emberjs/ember.js/blob/ee8ab348484421f254fe995c6f04f0d4a0859cfa/packages/ember-routing/lib/system/generate_controller.js#L19 Works in the resolver b/c `useRouterNaming` is called for controllers https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/resolver.js#L288 barf.

File: packages/ember-routing/tests/system/controller_for_test.js [hunk 2]
```diff
@@ -1,112 +1,59 @@
...
+    return this.visit('/').then(() => {
+      let controller = generateController(this.applicationInstance, 'home');
+      assert.ok(controller instanceof Controller, 'should return controller');
     });
   }
-});
 
-QUnit.test('generateController should return Ember.Controller', function() {
-  let controller = generateController(appInstance, 'home');
+  ['@test generateController should return App.Controller if provided'](assert) {
```
Comment (mixonic) [inline packages/ember-routing/tests/system/controller_for_test.js]: Can this test be changed to `should return controller:basic if resolved` instead? The global `App.Controller` has no relevance to the remaining test (and is less informative).

File: packages/ember-routing/tests/system/controller_for_test.js [hunk 3]
```diff
@@ -1,112 +1,59 @@
...
+    });
+  }
 
-  if (EMBER_NO_DOUBLE_EXTEND) {
-    ok(controller instanceof BasicController, 'should return base class of controller');
-  } else {
-    let doubleExtendedFactory;
-    ignoreDeprecation(() => {
-      doubleExtendedFactory = appInstance._lookupFactory('controller:basic');
+  ['@test generateController should return controller:basic if provided'](assert) {
```
Comment (mixonic) [inline packages/ember-routing/tests/system/controller_for_test.js]: This test was poorly named. It was testing the case of a registered factory. The test above it tests the resolved case.

As such, this test should not use `this.add('controller:basic'` but instead should register the factory:

* `this.application.register('controller:basic'`
* Even more like the old test would be `this.applicationInstance.register('controller:basic'` after the `visit` call.

Finally the test name should be `@test generateController should return controller:basic if registered` (note "registered" vs. "resolved" in the above test).

**Review by mixonic — CHANGES_REQUESTED** [review]
I've requested some very small cleanup but this is pretty close!

File: packages/ember-routing/tests/system/controller_for_test.js (hunk 1 above)
Comment (mmun) [inline packages/ember-routing/tests/system/controller_for_test.js]: @mixonic Whoa. Didn't expect that!

File: packages/ember-routing/tests/system/controller_for_test.js [hunk 4]
```diff
@@ -1,112 +1,58 @@
...
 
-  if (EMBER_NO_DOUBLE_EXTEND) {
-    ok(controller instanceof BasicController, 'should return base class of controller');
-  } else {
-    let doubleExtendedFactory;
-    ignoreDeprecation(() => {
-      doubleExtendedFactory = appInstance._lookupFactory('controller:basic');
+  ['@test generateController should return controller:basic if registered'](assert) {
+    let BasicController = Controller.extend();
+    this.application.register('controller:basic', BasicController);
```
Comment (stevenwu) [inline packages/ember-routing/tests/system/controller_for_test.js]: @mixonic I tried `this.applicationInstance.register('controller:basic', BasicController)` after `visit` but the test fails.

**Review by mixonic — APPROVED** [review]

**Comment by mixonic** [comment]
Excellent, thank you @stevenwu!

**Pull request merged**